langchain/vectorstores/chroma.ts

```diff
@@ -0,0 +1,194 @@
+import fs from "fs/promises";
+import path from "path";
+import type {
+  ChromaClient as ChromaClientT,
+} from "chromadb";
+import { ChromaClient as ChromaClientO } from "chromadb";
+
+import { Embeddings } from "../embeddings/base";
+
+import { DocStore, SaveableVectorStore } from "./base";
+import { Document } from "../document";
+
+let ChromaClient: typeof ChromaClientT | null = null;
+
+try {
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  ({ ChromaClient } = require("chromadb"));
+} catch {
+  // ignore error
+}
+ChromaClient = ChromaClientO;
+
+export interface ChromaLibArgs {
+  space: string;
+  numDimensions?: number;
+  collection_name: string,
+}
+
+export class Chroma extends SaveableVectorStore {
+  index?: ChromaClientT;
+  
+  docstore: DocStore;
+
+  args: ChromaLibArgs;
+
+  collection_name: string;
+
+  constructor(
+    args: ChromaLibArgs,
+    embeddings: Embeddings,
+    docstore: DocStore,
+    index?: ChromaClientT,
+  ) {
+    super(embeddings);
+    this.index = index;
+    this.args = args;
+    this.embeddings = embeddings;
+    this.docstore = docstore;
+    this.collection_name = args.collection_name;
+  }
+
+  async addDocuments(documents: Document[]): Promise<void> {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    await this.addVectors(await this.embeddings.embedDocuments(texts), documents);
+  }
+
+  async addVectors(vectors: number[][], documents: Document[]) {
+    if (vectors.length === 0) {
+      return;
+    }
+    if (!this.index) {
+      if (this.args.numDimensions === undefined) {
+        this.args.numDimensions = vectors[0].length;
```

Comment:
If chromadb doesn't use this arg then Im not against just removing it, unless you think the error messages that compare it to the embedding length are worth it

Reply:
Feels like a little local validation that keeps it instep with the `hnsw` implementation is nice. We do more validation on the JS client side as well.